### ISSUE TITLE
Route Producer introduces delay

### ISSUE THREAD

**Issue by ianshade** [CasparCG/server#1248]
### Expected behaviour
There should be 0 frames of delay when routing layers between channels and within a channel, and there should be 0 frames (or maximally 1 frame in specific cases) of delay when routing channels to other channels.

### Current behaviour
The Route Producer currently introduces delay even when routing layers on the same channel. Routing a layer to a layer on the same channel adds 1 frame of delay. Routing a layer to another channel and routing that channel back adds 3 frames of delay. 

### Solution suggestion
Routing a layer seems obvious: a frame comes from a producer on layer n; it is uploaded to the GPU and is available to the mixer when drawing layer n, so it is also available to be drawn on layer m where we routed layer n.
When routing a layer to another channel it also should work fine assuming that drawing of corresponding frames of all channels happens after producers' frames for all layers on all channels have been uploaded.
When routing a channel to another channel, corresponding frames of those channels should be drawn in order that allows drawing one channel on another. If we routed channel m to channel n then channel m should be drawn first and that texture should be used when drawing channel n. 
It becomes more complex with circular dependencies and that might require some buffering hence the maximal 1 frame of delay, but I don't see many cases when one would like to route channel n to channel m and channel m back to channel n at the same time.

----

### Steps to reproduce

1. Setup two channels with the same framerate
2. `MIXER 1 GRID 2` (for convenient display)
3. `PLAY 1-1 SOMEVIDEO`
4. `PLAY 1-2 route://1-1` (routing layer 1 to the same channel)
5. `PLAY 2-1 route://1-1` (routing layer 1 to channel 2)
6. `PLAY 1-3 route://2` (routing channel 2 back to channel 1)

### Environment
* Commit: 4a8f03f
* Server version: 2.3
* Operating system: Windows 10

---

### Screenshots
https://imgur.com/a/Xznms7p

**Comment by PeterC89** [comment]
Duplicate of #815 (ish)

**Comment by Julusian** [comment]
This is not a simple thing to resolve in the current architecture.
Each channel runs in parallel on its own clock, so there is no guarantee of the source channel producing the frames before the route tries to consume it

**Comment by premultiply** [comment]
Maybe a command to instantaneous MOVE the content of layer or channel to somewhere else would help to avoid the buffering between async channels for some cases.
At the end this would work like a preset and program bus on a vision mixer.

**Comment by Julusian** [comment]
This is resolved in 8998411317e8d80f6dd8c02e94158cbd954d1f50, with a maximum of 1 frame when doing cross-channel routes
